lib/ramble/ramble/language/shared_language.py

```diff
@@ -77,8 +79,57 @@ def _execute_figure_of_merit_context(obj):
     return _execute_figure_of_merit_context
 
 
+# For a FOM, the direction that is 'better' e.g., faster is better
+class BetterDirection(Enum):
+    HIGHER = 1
+    LOWER = 2
+    INDETERMINATE = 3  # requires interpretation or FOM type not defined
+    INAPPLICABLE = 4  # non-numerical or no direction is 'better', like strings or categories
+
+    def get_direction_str(self):
+        if self == BetterDirection.HIGHER:
+            return " (Higher is Better)"
+        if self == BetterDirection.LOWER:
+            return " (Lower is Better)"
+        else:
+            return ""
+
+
+class FomType(Enum):
+    TIME = 1
+    THROUGHPUT = 2
+    MEASURE = 3
+    CATEGORY = 4
+    INFO = 5
+    UNDEFINED = 6
+
+    def better_direction(self):
+        direction = {
+            FomType.TIME: BetterDirection.LOWER,
+            FomType.THROUGHPUT: BetterDirection.HIGHER,
+            FomType.MEASURE: BetterDirection.INDETERMINATE,
+            FomType.CATEGORY: BetterDirection.INAPPLICABLE,
+            FomType.INFO: BetterDirection.INAPPLICABLE,
+            FomType.UNDEFINED: BetterDirection.INDETERMINATE,
+        }
+
+        return direction[self]
+
+    def copy(self):
```

Comment:
I was trying to avoid making a special case out of FomType in this loop within ApplicationBase._analysis_dicts() so that it can copy the FomType object:
```
            for fom, fom_def in mod.figures_of_merit.items():
                fom_definitions[fom] = {"origin": f"{mod}", "origin_type": "modifier"}
                for attr in fom_def.keys():
                    if isinstance(fom_def[attr], (list, FomType)):
                        fom_definitions[fom][attr] = fom_def[attr].copy()
                    else:
                        fom_definitions[fom][attr] = self.expander.expand_var(
                            fom_def[attr], mod_vars
                        )
```

Do you know of a better way to do it?

Reply:
Updated to deepcopy to match the other class methods